src/backend/kms/surface/mod.rs

```diff
@@ -1205,13 +1205,23 @@ impl SurfaceThreadState {
                                     .into_iter()
                                     .flatten();
 
+                                // If the screen is rotated, we must convert damage to match output.
+                                let adjusted = damage.iter().copied().map(|mut d| {
+                                    d.size = d
+                                        .size
+                                        .to_logical(1)
+                                        .to_buffer(1, output_transform)
+                                        .to_logical(1, Transform::Normal)
+                                        .to_physical(1);
```

Comment:
Ouch, yeah this is ugly.

I think if would possibly a bit cleaner to transform this already in line 1006 https://github.com/pop-os/cosmic-comp/pull/986/files#diff-85c7861d0290a1e31c47e8e1d31b15f37d2bb03a1b4edb046d9173eef8c9f758R1006 .
But given this is quite a hack anyway, we don't have to bother.